Issue 3234: Speedup Load Time

ericchristoffersen opened the pull request:
Implement regex/hash based string substitution object to perform multiple substitutions in 2 passes. Speeds up athlete data load by 2x.

amtriathlon commented:
TBH at first sight I didn't liked the added complexity but I gave it a try: my full history athlete with +6000 activities, which took a full minute to load the cache in a Windows 10 notebook with SSD and 16GB RAM, now loads it in around 30".

liversedge commented:
Crikey, had not realised how terribly slow this aspect was, I definiteley upvote this fixup !

ericchristoffersen commented:
Like shooting fish in a barrel. I changed to use QStringRef for substring before replace which was another 20% speedup. Added string handling cleanup to ridecache::save for 2x speedup to app exit.
Removed ifdefs. I think I'm done now with dumb string fixes.
